### PR TITLE
[IMP] (website_)sale*: fixes and improvements

### PR DESCRIPTION
**Pull request by vava-odoo**
This commit simplifies the methods to add context values in `website_sale_*` brigde modules and removes the duplicate on sale order line of the shop warning for stock.

task-3074645

See also:
- https://github.com/odoo/enterprise/pull/36197
- https://github.com/odoo/upgrade/pull/4686

### CODE REVIEW

File: addons/website_sale_stock_product_configurator/views/product_configurator_templates.xml [hunk 1]
```diff
@@ -1,9 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo>
     <template id="website_sale_stock_modal" inherit_id="website_sale_product_configurator.product_quantity_config" name="Stocks Modal">
-        <xpath expr="//input[@type='text'][hasclass('quantity')]" position="attributes">
-          <attribute name='t-att-data-max'>max(product.sudo()._get_max_available_qty(), 1) if handle_stock and product.type == "product" and not product.allow_out_of_stock_order else None</attribute>
-        </xpath>
```
Comment (chevalierv) [inline addons/website_sale_stock_product_configurator/views/product_configurator_templates.xml:6]: Why?

**Review by chevalierv — CHANGES_REQUESTED** [review]

File: addons/website_sale_stock_product_configurator/views/product_configurator_templates.xml (hunk 1 above)
Comment (vava-odoo) [reply addons/website_sale_stock_product_configurator/views/product_configurator_templates.xml]: `handle_stock` is always set to `False` in the controller, so this xpath is useless. But you're right, not related to this task, I will fix that in another PR.

**Review by chevalierv — APPROVED** [review]

**Review by Feyensv — COMMENTED** [review]
(seems i forgot to post the community comments :innocent: )

File: addons/website_sale_stock/controllers/main.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ def shop_payment_transaction(self, *args, **kwargs):
             if line.product_id.type == 'product' and not line.product_id.allow_out_of_stock_order:
                 cart_qty, avl_qty = order._get_cart_and_free_qty(line=line)
                 if cart_qty > avl_qty:
-                    line._set_shop_warning_stock(cart_qty, max(avl_qty, 0))
+                    line.shop_warning = order._set_shop_warning(cart_qty, max(avl_qty, 0))
```
[the remark this answers is not in the document]
Comment (vava-odoo) [reply addons/website_sale_stock/controllers/main.py]: I wanted to get rid of the duplicated method (same in sale order and sale order line). But I guess you're right.

File: addons/website_sale_mondialrelay/models/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
```
Comment (Feyensv) [inline addons/website_sale_mondialrelay/models/sale_order.py:9]:
```suggestion
    _inherit = 'sale.order'
```
nitpicking :D

**Review by Feyensv — APPROVED** [review]

File: addons/sale_stock/models/sale_order_line.py [hunk 4]
```diff
@@ -43,7 +43,8 @@ def _compute_qty_to_deliver(self):
 
     @api.depends(
         'product_id', 'customer_lead', 'product_uom_qty', 'product_uom', 'order_id.commitment_date',
-        'move_ids', 'move_ids.forecast_expected_date', 'move_ids.forecast_availability')
+        'move_ids', 'move_ids.forecast_expected_date', 'move_ids.forecast_availability',
+        'warehouse_id')
```
Comment (Whenrow) [inline addons/sale_stock/models/sale_order_line.py:47]: I would remove `rental` in the commit message as it is also useful in sale_stock in general

**Review by Feyensv — APPROVED** [review]
@robodoo delegate+

**Comment by vava-odoo** [comment]
@robodoo r+

**Comment by vava-odoo** [comment]
@robodoo rebase-ff

**Comment by vava-odoo** [comment]
@robodoo r+

**Pull request closed without merging**